encoding.bs

```diff
@@ -995,7 +971,34 @@ steps:
 <p>To <dfn export>UTF-8 encode</dfn> a scalar value stream <var>stream</var>, return the result of
 <a lt=encode for=/>encoding</a> <var>stream</var> using encoding <a>UTF-8</a>.
 
+<hr>
+
+<p>To <dfn export>BOM sniff</dfn> a byte stream <var>stream</var>, run these steps:
+
+<ol>
+  <li><p>Wait until <var>stream</var> has three bytes available or the <a>end-of-stream</a> has been
+  reached, whichever comes first.
+
+  <li>
+    <p>For each of the rows in the table below, starting with the first
+    one and going down, if <var>stream</var> <a for="byte sequence">starts with</a>
+    the bytes given in the first column, return the <a for=/>encoding</a> given
+    in the cell in the second column of that row. Do not consume those bytes.
+
+    <table>
+    <tbody><tr><th>Byte order mark<th>Encoding
+    <tr><td>0xEF 0xBB 0xBF<td><a>UTF-8</a>
+    <tr><td>0xFE 0xFF<td><a>UTF-16BE</a>
+    <tr><td>0xFF 0xFE<td><a>UTF-16LE</a>
+    </table>
+
+  <li><p>Otherwise, return null.
+</ol>
 
+<p class=note>This hook is a workaround for the fact that <a>decode</a> has no way to communicate
+back to the caller that it has found a byte order mark and is therefore not using the provided
+encoding. The hook is to be invoked before <a>decode</a>, and it will return an encoding
+corresponding to the byte order mark found, or null otherwise.
```

Comment:
You need to add some newlines here to preserve the original number of newlines before a heading.

Reply:
There are two newlines here in the original, but elsewhere in the spec there are three newlines before an `<h2>`.

Reply:
I count three unchanged blank lines in the diff.

Reply:
My bad, I was doing `git diff` with respect to origin/master and it was matching one of the blank lines in the original in the middle of my change.